File: src/main/java/engineer/carrot/warren/thump/util/helper/PredicateHelper.java
package engineer.carrot.warren.thump.util.helper;

import com.google.common.base.Predicate;

public class PredicateHelper {
    public static class StartsWithPredicate implements Predicate<String> {
        private String matcher;

        public StartsWithPredicate(String matcher) {
            this.matcher = matcher;
        }

        @Override
        public boolean apply(String input) {
            return input.startsWith(matcher);
        }

        @Override
        public boolean equals(Object object) {
            return this.equals(object);
        }
    }

    public static class DoesNotContainPredicate implements Predicate<String> {
        private String matcher;

        public DoesNotContainPredicate(String matcher) {
            this.matcher = matcher;
        }

        @Override
        public boolean apply(String input) {
            return !input.contains(matcher);
        }

        @Override
        public boolean equals(Object object) {
            return this.equals(object);
        }
    }
}

File: src/main/kotlin/engineer/carrot/warren/thump/helper/PredicateHelper.kt
package engineer.carrot.warren.thump.helper

import com.google.common.base.Predicate

class PredicateHelper {
    class StartsWithPredicate(private val matcher: String) : Predicate<String> {

        override fun apply(input: String?): Boolean {
            return input!!.startsWith(matcher)
        }

        override fun equals(other: Any?): Boolean {
            return this == other
        }

        override fun hashCode(): Int{
            return matcher.hashCode()
        }
    }

    class DoesNotContainPredicate(private val matcher: String) : Predicate<String> {

        override fun apply(input: String?): Boolean {
            return !input!!.contains(matcher)
        }

        override fun equals(other: Any?): Boolean {
            return this == other
        }

        override fun hashCode(): Int{
            return matcher.hashCode()
        }
    }
}
